src/t8_element_cxx.hxx

```diff
@@ -634,6 +634,55 @@ public:
+   *  Depending on the subelement type, the number of subelements, 
+   *  to fill the parent element, can differ.
+   *  \param [in] elem A valid element
+   *  \param [in] type The subelement type
+   *  \param [out] c An array of all children subelements of the parent quad element elem
+   */
+  virtual void        t8_element_to_subelement (const t8_element_t * elem,
+                                                t8_element_t * c[],
+                                                int type) = 0;
+
+  /** This function determines the vertex coordinates of subelements.
+   *  \param [in] elem A valid subelement 
+   *  \param [in] vertex the number of the vertex, whose coordinates should be determined
+   *  \param [out] c An array, whose entries will be filled with the coordinates of the
+   *                 subelement. 
+   * Note, that subelements can have another number of vertecies, compared to the used
+   * eclass scheme. For example, subelements, that remove hanging nodes from the quad scheme,
+   * are triangles with 3 instead of 4 verticies.           
+   */
+  virtual void        t8_element_vertex_coords_of_subelement (const t8_element_t * t,
+                                                              int vertex, int coords[]) = 0; 
+
+  /** This function will determine the number of children subelements, depending on the 
+   *  subelement type. 
+   *  \param [in] subelement_type The subelement type as an integer
+   *  \return the number of subelements, a parent element is split into, as an integer.
+   *  This function can be used, to allocate enough memory before transforming an element
+   *  into subelements.  
+   */
+  virtual int         t8_element_get_number_of_subelements (int subelement_type) = 0;  
+
+  /** This function will determine the location of a specific subelement in the parent element.
+   *  Since different subelement types are possible, it is apriori not known, where for example the
+   *  subelement with id 3 is located. 
+   *  \param [in] elem A valid subelement
+   *  \param [out] An array, whose entries will be filled with location information of the subelement 
+   *  All information in the location array will be needed to determine the verticies of any subelement.
+   */
+  virtual void         t8_element_get_location_of_subelement (const t8_element_t * elem, int location[]) = 0;
+
+  /** We allow, that subelements can have a different shape, than the given eclass scheme. 
+   *  For example, subelements, that remove hanging nodes from the quad scheme, are triangles.
+   *  \param [in] elem A valid element
+   *  \return the shape of the element. If it is not a subelement, the shape equals the shape 
+   *  of the given eclass. Else, the shape can differ an is dependend on the specific case.  
+   */
+  virtual t8_element_shape_t t8_element_get_shape (const t8_element_t * elem) = 0;                                               
```

Comment:
I removed this function from this file and all other schemes as well since we dont need it anymore. I also removed the implementations from all other subelement related functions from the schemes (like hex_cxx.cxx and .hxx). An implementation of the subelement functions can be found in common_cxx.cxx and .hxx and will abort the code there if the subelement function is not implemented for the specific scheme (which is the case for all schemes except quad_w_sub).